src/tree-root.tsx

```diff
@@ -14,17 +14,15 @@ export type TreeRootProps<BranchType, LeafType> =
     data: TreeData<BranchType, LeafType>;
     depth?: number;
     renderLeaf: (item: LeafType) => ReactNode;
-    renderBranch: (
-      item: BranchType,
-      isToggled: boolean,
-      onToggle: () => void,
-    ) => ReactNode;
+    renderBranch: (item: BranchType) => ReactNode;
+    getIsToggled: (data: TreeData<BranchType, LeafType>) => boolean;
```

Comment:
I would prefer this become a fully controlled element with this being passed in as data

Reply:
Added a `isToggled` property to `TreeData` and removed the whole `getIsToggled(...)` function, additionally, had to change the parameter of `renderBranch` to be the `tree` instead of just the `tree.data` so `isToggled` can be properly accessed on `renderBranch`